reader.go

```diff
@@ -73,52 +75,77 @@ func (r *Reader) addCurrentRoutingNumberSummary(rns *RoutingNumberSummary) {
 }
 
 // NewReader returns a new ACH Reader that reads from r.
-func NewReader(r io.Reader) *Reader {
+func NewReader(r io.Reader, opts ...ReaderOption) *Reader {
 	f := NewFile()
 	f.Control = FileControl{}
-	return &Reader{
-		File:    *f,
-		scanner: bufio.NewScanner(r),
+	reader := &Reader{
+		File:       *f,
+		scanner:    bufio.NewScanner(r),
+		decodeLine: Passthrough,
+	}
+	for _, opt := range opts {
+		opt(reader)
 	}
+	return reader
 }
 
-// SetFormat of imagecashletter file
-func (r *Reader) SetFormat(format Format) {
-	switch format {
-	case Discover:
-		r.format = format
-	case DSTU:
-		r.format = format
-		r.scanner.Split(scanVariableLengthLines)
-	}
+// DecodeLineFn is used to decode a scanned line into desired encoding.
+// Depending on X9 spec, cashletter could be encoded as ASCII or EBCDIC
+type DecodeLineFn func(lineIn string) (lineOut string)
+
+// Passthrough will return line as is
+func Passthrough(lineIn string) (lineOut string) {
+	return lineIn
 }
 
-// ScanVariableLengthLines will return lines from imagecashletter file based on encoded line length
-// This implements bufio.SplitFunc
-func scanVariableLengthLines(data []byte, atEOF bool) (advance int, token []byte, err error) {
-	// nothing to scan
-	if atEOF && len(data) == 0 {
-		return 0, nil, nil
+// DecodeEBCDIC will decode a line from EBCDIC-0037 to UTF-8
+func DecodeEBCDIC(lineIn string) (lineOut string) {
+	lineOut, err := encoding.EBCDIC.NewDecoder().String(lineIn)
+	if err != nil {
+		fmt.Printf("Error decoding '%X' as EBCDIC: %v\n", lineIn, err)
 	}
+	return lineOut
+}
 
-	// line length can be variable
-	// use the 4 control bytes at the beginning of a line to determine its length
-	ctrl := data[0:4]
-	dataLen := int(binary.BigEndian.Uint32(ctrl))
-	lineLen := 4 + dataLen
+// ReaderOption can be used to change default behavior of Reader
+type ReaderOption func(*Reader)
 
-	// the last calculated line is expected to match the remaining bytes
-	if atEOF && lineLen != len(data) {
-		return len(data), data, io.ErrUnexpectedEOF
+//ReadVariableLineLengthOption allows Reader to split imagecashletter files based on encoded line lengths
+func ReadVariableLineLengthOption() ReaderOption {
```

Comment:
According to the spec:
> Each record in the X9.100-187 standard is defined as variable length and shall be preceded by a four (4) byte field known as the Inserted Length Field.

If each record is supposed to be preceded by Inserted Length Field should this be set as the default reader option in `NewReader()` (reader.go:78)? Thoughts @jenakan and @adamdecaf?

Reply:
I'm thinking either one of the following (in order of preference):
1. Update files.go:106 to create a `NewReader()` with `ReadVariableLineLengthOption`, **or**
2. Update `NewReader()` to assume `ReadVariableLineLengthOption` if no options were passed in

Reply:
The 4 byte field is required on X9.100-187 and X9.37-2003. I haven't come across one that used LF/CR.

Not sure how widely this library is in use, but a change to the default reader would be considered breaking change.

Reply:
We're in pre-v1 release territory, so I'm okay with breaking default behaviour on a minor version if we support the old (as needed) to better serve the industry we're aware of.

Reply:
Makes sense. The reason I ask is because the x9 spec makes it sound like a file is not a valid x9 file if each record doesn't contain these four bytes. That being the case, wouldn't the desired default behavior be the one that's in line with the spec? (genuine question here, not rhetorical...)

The spec also seems to indicate that any agreements between parties can supersede the spec, so if there's a "norm" in the industry that is contrary to my suggestion please do share what you know. :)

Reply:
> That being the case, wouldn't the desired default behavior be the one that's in line with the spec?

Yes. We want our OSS libraries to follow the spec. Deviations from vendors or old revisions are to be configured as non-default code paths.

Reply:
@jenakan Let's move forward with this PR as is since the default reader behavior isn't really in scope for this PR. I'll create a new PR where we can discuss this further. Again, thank you for these contributions.